lib/src/components/NxSidebarNavigation/types.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import * as PropTypes from 'prop-types';
+import { WeakValidationMap } from 'react';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+
+export interface SidebarNavLinkProps {
+  name: string;
+  href: string;
+  icon: IconProp;
+  current?: boolean | null;
+}
+
+export interface Props {
```

Comment:
It's generally preferred to base the stateful component props on the stateless ones using `extends` and potentially [`Omit`](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys)

Reply:
> It's generally preferred to base the stateful component props on the stateless ones using extends and potentially Omit

And similarly, to base the stateful propTypes on the stateless ones using object spread and ramda's `omit`.

Reply:
> > It's generally preferred to base the stateful component props on the stateless ones using extends and potentially Omit
> 
> And similarly, to base the stateful propTypes on the stateless ones using object spread and ramda's `omit`.

Done in 0f25e721c9505e11759b66031eb242387ef74dd1